### PR TITLE
Abstract the code

### PR DESCRIPTION
**Pull request by dbhynds**
Refactors a bunch of things to from the other PR to make separation of concerns cleaner and naming consistent.

A diff with `master` can be viewed here: https://github.com/packbackbooks/lti-1-3-php-library/compare/master..mgmt-132-migration-abstraction

### CODE REVIEW

File: src/LtiMessageLaunch.php [hunk 1]
```diff
@@ -118,52 +120,27 @@ public static function fromCache(
...
-
-        return false;
-    }
-
-    private function lti1p1InstallationSignatureMatches(Lti1p1Installation $ltiInstall, array $launchData): bool
+    public function initialize(array $request)
     {
-        // Check to see if we have params to calculate oauth_consumer_key_sign
-        if (!isset($launchData[LtiConstants::LTI1P1]) || !isset($launchData[LtiConstants::LTI1P1]['oauth_consumer_key_sign'])) {
-            throw new LtiException(static::ERR_MISSING_OAUTH_CONSUMER_KEY_SIGN);
```
Comment (erikdonohoo) [inline src/LtiMessageLaunch.php:156]: The reason I distinguished between not having the key_sign passed, and the key_sign not matching, was so we could give different messages to a user who is attempting this launch, as there will be two different things needed to be done in order to fix the issue. if they simply don't have the key_sign, then the deployment needs to verified manually. And they should not be able to proceed as the deployment was missing. But if we only rely on the "deployment is missing" error, I can't know that it was an attempted migration. The other error is more significant in that it had the proper things to migrate automatically, but the key sign didn't match. meaning the LMS is doing something wrong with their 1.3 launch and something needs to be rectified between Platform and Tool.

File: src/LtiMessageLaunch.php [hunk 2]
```diff
@@ -118,52 +120,27 @@ public static function fromCache(
...
-
-        if (!$ltiInstall->compareOauthConsumerKeySign($launchData, $signature)) {
-            throw new LtiException(static::ERR_OAUTH_CONSUMER_KEY_SIGN_MISMATCH);
-        }
-
-        return true;
+        return $this->setRequest($request)
+            ->validate()
+            ->migrate()
+            ->cacheLaunchData();
```
Comment (erikdonohoo) [inline src/LtiMessageLaunch.php:143]: Doesn't `validate` already `cacheLaunchData()`? @dbhynds ?

**Comment by erikdonohoo** [comment]
Hey @dbhynds left you some feedback on this. Waiting to hear back from you

File: src/LtiMessageLaunch.php (hunk 2 above)
Comment (dbhynds) [reply src/LtiMessageLaunch.php:143]: It does, but I want to remove it from there in the next major version. IMO people should either do:
```php
$launch = LtiMessageLaunch::new()->initialize($request);

// Or manually by doing this (if they wanted to omit migrations or some other step):
$launch = LtiMessageLaunch::new()->setRequest($request)
    ->validate()
    ->cacheLaunchData();
```

File: src/LtiMessageLaunch.php (hunk 1 above)
Comment (dbhynds) [reply src/LtiMessageLaunch.php:156]: So then, should the `if` statement here actually be this? ...
```php
if (isset($launchData[LtiConstants::LTI1P1]) && !isset($launchData[LtiConstants::LTI1P1]['oauth_consumer_key_sign']))
```

That is, we have data for an LTI 1.1 migration, but they didn't send over the sign key

File: src/LtiMessageLaunch.php (hunk 2 above)
Comment (erikdonohoo) [reply src/LtiMessageLaunch.php:143]: That sounds fine then

File: src/LtiMessageLaunch.php (hunk 1 above)
Comment (dbhynds) [reply src/LtiMessageLaunch.php:156]: I've got some open questions related to this with IMS Global, but I'll make sure different error messages are included in the final PR. I'm gonna merge this for now, tho.

**Pull request merged**